### DISCUSSION TITLE
EW11 connection won't work

### DISCUSSION THREAD

**Discussion by jodewee** [wlcrs/huawei_solar#247] [Setting up the integration]
### Describe the issue

Hi, 
i connected the cables as described, but do net get any info into the modbus gateway.
Do you have any screenshots available of the EW11 config?
![image](https://user-images.githubusercontent.com/40162805/211159169-ca842d75-d0a8-43eb-9dfb-76cec0d32cf9.png)
![image](https://user-images.githubusercontent.com/40162805/211159199-05458e62-95b5-49b8-98a0-8d24d4d35ca8.png)
![image](https://user-images.githubusercontent.com/40162805/211159245-f68aecb9-21b3-49f6-b572-ee87d36c60d8.png)

### Bescribe your Huawei Solar Setup

Inverter Type:
Inverter Firmware version:
SDongle present: yes / no
Power meter present: three phase / single phase / no
Battery: LG RESU xxkWh / LUNA2000 xxkWh / None
Battery Firmware version:

### How do you connect to the inverter?

Over serial, with a serial-to-USB stick

### Upload your Diagnostics File

Drag & Drop your Diagnostics File here.

### Upload your relevant debug logs

```text
/
connect to inverter failed
```

### Please confirm the following:

- [X] The problem is still present in the latest release of this integration.
- [X] I did not find an existing issue describing this problem.
- [X] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [ ] I increased the logging configuration, restarted HA, and have copied all relevant log lines from the 'Full Logs' into the textarea.

**Comment by jodewee** [comment]
error: 
Reading from the inverter failed.

**Comment by jodewee** [comment]
![image](https://user-images.githubusercontent.com/40162805/211159536-bab865c5-30db-4784-8fef-2832bd012f58.png)

**Comment by jodewee** [comment]
Logger: huawei_solar.huawei_solar
Source: /usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py:129
First occurred: 16:54:57 (3 occurrences)
Last logged: 16:59:03

Aborting client creation due to error.
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 280, in _do_read
    response = await self._client.protocol.read_holding_registers(
  File "/usr/local/lib/python3.10/site-packages/pymodbus/client/base.py", line 295, in execute
    resp = await asyncio.wait_for(req, timeout=self.params.timeout)
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 445, in wait_for
    return fut.result()
pymodbus.exceptions.ConnectionException: Modbus Error: [Connection] Connection lost during request

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 123, in create
    await huawei_solar._initialize()
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 85, in _initialize
    self.time_zone = (await self.get(rn.TIME_ZONE)).value
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 198, in get
    return (await self.get_multiple([name], slave))[0]
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 233, in get_multiple
    response = await self._read_registers(registers[0].register, total_length, slave)
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 313, in _read_registers
    result = await _do_read()
  File "/usr/local/lib/python3.10/site-packages/backoff/_async.py", line 151, in retry
    ret = await target(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 309, in _do_read
    raise ReadException(message) from err
huawei_solar.exceptions.ReadException: Could not read register value, has another device interrupted the connection?

**Comment by jodewee** [comment]
as a fact, i think i do have the s-dongle installed to

**Reply by wlcrs** [reply]
As stated on the Wiki: ⚠Warning⚠: The SDongleA-05 dongle interferes with devices connected to the RS485A1 and RS485B1 pins on the inverter. If your inverter has such a dongle installed, we recommend connecting to the Modbus-TCP port exposed by that dongle instead.

**Reply by henfri** [reply]
I am having the same issue and can confirm that s-dongle and Modbus-RTU does not work (interfere is for my case not a strong enough expression).

@wlcrs  Is there any way to use RTU and the s-dongle at the same time?

**Reply by wlcrs** [reply]
No

**Comment by henfri** [comment]
Ok, thanks.
